### ISSUE TITLE
Deltarune crashes VirtIO-FS service at start

### ISSUE THREAD

**Issue by viktor-prutyanov** [virtio-win/kvm-guest-drivers-windows#724]
> Please give examples of "some other programs"

I have Deltarune, which is probably the only game i have on the shared that i can run on the vm at the moment but i could attempt to install notepad++ to the shared folder and check that as well. 
but when i ran deltarune there wasnt any obvious error code it just exited out and the service died the same as before.

_Originally posted by @GrandtheUK in https://github.com/virtio-win/kvm-guest-drivers-windows/issues/723#issuecomment-1022304628_

**Comment by viktor-prutyanov** [comment]
Reproduced with virtio-win at https://github.com/virtio-win/kvm-guest-drivers-windows/commit/5fcd9ea620559e440ba4841b3a190fa6f9a22f15 and DELTARUNE Demo 1.0.7.0 (from Steam) installed on VirtIO-FS.

**Comment by GrandtheUK** [comment]
installing notepad++ to the virtio-fs had a few errors during the install but running it works just fine. so i'm not entirely sure whats going on.

**Comment by GrandtheUK** [comment]
I got the following in event viewer on one of the times that the service died if it is useful
![Screenshot_20220127_200014](https://user-images.githubusercontent.com/31855376/151463843-1713f755-0718-4cff-8034-a4e4dd7187cb.png)

**Comment by arthurrasmusson** [comment]
I have created a VM and placed the SteamLibrary folder on the VirtIO-FS drive. Upon attempting to install software from Steam to the VirtIO-FS drive immediately Steam throws "drive write error". I have also noticed attempting to copy or modify large files also results in failures such as the file becoming unreadable with a file named **?????** and/or the `virtiofs.exe` Windows guest service crashing.

**Comment by xiagao** [comment]
> I have created a VM and placed the SteamLibrary folder on the VirtIO-FS drive. Upon attempting to install software from Steam to the VirtIO-FS drive immediately Steam throws "drive write error". I have also noticed attempting to copy or modify large files also results in failures such as the file becoming unreadable with a file named "**?????**" and/or the `virtiofs.exe` Windows guest service crashing.

There is an internal bug that virtiofs.exe quit while doing iozone test, seems they are the same.
iozone.exe -az -g 1g -y 32k -i 0 -i  1 -f Z:\zzz -b C:\yyy
@viktor-prutyanov bz number is 2024518 ,could you have a check? Thank you.

**Comment by arthurrasmusson** [comment]
@xiagao if I wanted to build kvm-guest-drivers-windows from source to get the most recent fixes is there any recent commits related to this issue worth trying?

**Comment by YanVugenfirer** [comment]
Hello All,

Please help us understanding you use cases for using virtio-fs, and thus make us virtio-fs support better.
Please participate in the discussion and add your use cases: https://github.com/virtio-win/kvm-guest-drivers-windows/discussions/726

Thanks a lot,
Yan.

**Comment by xiagao** [comment]
> @xiagao if I wanted to build kvm-guest-drivers-windows from source to get the most recent fixes is there any recent commits related to this issue worth trying?

Sorry,this issue isn't resolved by now.:(

**Comment by arthurrasmusson** [comment]
@xiagao It's okay. If there's something I can do to help let me know.

**Comment by arthurrasmusson** [comment]
@YanVugenfirer Some of that is discussed here:
https://www.youtube.com/watch?v=xbVKMQ1Rz2Y

TL;DR: [LibVF.IO](https://libvf.io/) is working on hared filesystem between host and guest for WINE-like functionality in a GPU accelerated VM.

**Comment by savchenko** [comment]
Might be related: after the video type was changed (once!) to VirtIO/OpenGL-enabled Spice, `virtiofs.exe` keeps crashing on transfers. Windows file dialogue shows error 0x800703E3:

![image](https://user-images.githubusercontent.com/300146/163939380-83ae74fa-792d-4214-b29b-7e9914fb8e8e.png)

Reverting back to QXL does not help.

**Comment by YanVugenfirer** [comment]
@savchenko By any chance is the crash dump for the process created?

**Comment by savchenko** [comment]
VM in question is destroyed, but I'll try to reproduce and provide some additional information.

**Comment by YanVugenfirer** [comment]
@savchenko Thanks a lot!

**Comment by YanVugenfirer** [comment]
@savchenko BTW: I suggest to open a new issues with a description of a problem as it is a different from the original one.

**Comment by savchenko** [comment]
@YanVugenfirer turns out it has nothing to do with GPU. I was able to reproduce it reliably, details are in the https://github.com/virtio-win/kvm-guest-drivers-windows/issues/760
